service/sendgrid/sendgrid.go

```diff
@@ -0,0 +1,70 @@
+package sendgrid
+
+import (
+	"net/http"
+
+	"github.com/pkg/errors"
+
+	"github.com/sendgrid/sendgrid-go"
+	"github.com/sendgrid/sendgrid-go/helpers/mail"
+)
+
+// SendGrid struct holds necessary data to communicate with the SendGrid API.
+type SendGrid struct {
+	client            *sendgrid.Client
+	senderAddress     string
+	senderName        string
+	receiverAddresses []string
+}
+
+// New returns a new instance of a SendGrid notification service.
+// You will need a SendGrid API key.
+// See https://sendgrid.com/docs/for-developers/sending-email/api-getting-started/
+func New(apiKey, senderAddress, senderName string) *SendGrid {
+	client := sendgrid.NewSendClient(apiKey)
+
+	return &SendGrid{
+		client:            client,
```

Comment:
nit: could drop client variable and assign directly here.

Reply:
Thanks for pointing this out, applied your suggestion on a new commit